src/WhatToCook.Application/Services/MealPlanningService.cs

```diff
@@ -20,16 +20,22 @@ public MealPlanningService(IRecipesRepository recipesRepository, IMealPlanningRe
 
     public async Task<PlanOfMeals> Create(PlanOfMealRequest planOfMealRequest)
     {
-        var RecipeForMealPlan = planOfMealRequest.Recipes.Select(x => x.Name);
-        var recipes = _recipesRepository.GetRecipesByNameForMealPlan(RecipeForMealPlan);
+        var requestedRecipeIds = planOfMealRequest.Recipes.SelectMany(x => x.RecipeIds).ToList();
+        var recipes = _recipesRepository.GetRecipesByIdForMealPlan(requestedRecipeIds);
+        var dayRecipePairs = planOfMealRequest.Recipes
+            .SelectMany(dayRecipe => dayRecipe.RecipeIds,
+                        (dayRecipe, recipeId) =>
+                        new RecipePerDay(dayRecipe.Day, recipes.First(r => r.Id == recipeId))).ToList();
 
         var planOfMeals = new PlanOfMeals
     (
         planOfMealRequest.Name,
         DateTime.SpecifyKind(planOfMealRequest.FromDate, DateTimeKind.Utc),
         DateTime.SpecifyKind(planOfMealRequest.ToDate, DateTimeKind.Utc),
-        recipes
+        dayRecipePairs
+
     );
+        _logger.LogInformation($"Creating a meal plan with {dayRecipePairs.Count} recipes.");
```

Comment:
Could be done better by providing data to format instead of using interpolation.
Example:
`_logger.LogInformation($"Creating a meal plan with {dayRecipePairs.Count} recipes.");`

Can be changed to:
`_logger.LogInformation("Creating a meal plan with {numberOfRecipes} recipes", dayRecipePairs.Count);
`